map/traffic_manager.cpp

```diff
@@ -315,17 +319,22 @@ void TrafficManager::OnTrafficDataResponse(traffic::TrafficInfo const & info)
 
   // Update cache.
   size_t constexpr kElementSize = sizeof(traffic::TrafficInfo::RoadSegmentId) + sizeof(traffic::SpeedGroup);
-  size_t const dataSize = info.GetColoring().size() * kElementSize;
+
+  // Note. It's necessary to multiply by two because routing and rendering use separate caches.
+  size_t const dataSize = 2 * info.GetColoring().size() * kElementSize;
```

Comment:
I dislike the idea that traffic manager knows about routing caches and depends on them in its calculations. It's a bad thing, as it's quite easy to change something in routing and completely break this logic.